pacman/operations/routing_info_allocator_algorithms/zoned_routing_info_allocator.py

```diff
@@ -180,3 +245,48 @@ def __bits_needed(size):
         :rtype: int
         """
         return int(math.ceil(math.log(size, 2)))
+
+
+def flexible_allocate(machine_graph, n_keys_map):
+    """
+    Allocated with fixed bits for the Application/Paritition index but
+    with the size of the atom and machine bit changing
+
+    :param MachineGraph machine_graph:
+        The machine graph to allocate the routing info for
+    :param AbstractMachinePartitionNKeysMap n_keys_map:
+        A map between the edges and the number of keys required by the
+        edges
+    :rtype: tuple(RoutingInfo,
+        dict((ApplicationVertex, str), BaseKeyAndMask))
+    :raise PacmanRouteInfoAllocationException:
+    """
+    # check that this algorithm supports the constraints put onto the
+    # partitions
+
+    allocator = ZonedRoutingInfoAllocator(machine_graph, n_keys_map, True)
+
+    allocator._calculate_zones()
```

Comment:
same issue

Reply:
There are two ways of doing it that differ so  minorly that just passing in a different flag tot he init is enough.

Pacman execution engine does not support changing param nor static methods but it does support file level methods which can then add the extra flag.